---
title: "Plotting Heatmaps"
output: html_document
---

## Load our libraries for plotting trees
```{r setup, include=TRUE,echo=TRUE}
knitr::opts_chunk$set(echo = TRUE)


#load the required packages
library(heatmap.plus)
library(gplots)
library(ggplot2)

#load color definitions from a separate file so you only have to change them in one place
source("../figure_colors.r")


```

## Load a gene presence/absence table and strain annotation table

```{r loaddata, include=TRUE,fig.width=12,fig.height=12}


#load a table of Ti plasmid phenotypes
plasmidtable <- read.table("./plasmidtypes.txt", sep = "\t", header=T, stringsAsFactors = F, quote = "")


#load a table with our gene presence/absence information for each strain
pangenome <- read.table("./pangenome_matrix_t0_renamed.names.tab", sep = "\t", header=T, stringsAsFactors = F)
row.names(pangenome) <- pangenome$strain
pangenome$strain <- NULL

#convert values above 1 (multi-copy) to 1 (gene presence)
pangenome[pangenome > 0] <- 1

#select only genes found in 3 or more plasmids to simplify
pangenome <- pangenome[, which(colSums(pangenome) > 2)]



```



## Plot a heatmap representation of gene presence/absence

```{r plotheatmap, include=TRUE,fig.width=12,fig.height=12}

#cluster plasmids (rows) based on gene ortholog content
row_distance = dist(pangenome, method = "binary")
row_cluster = hclust(row_distance, method = "ward.D2")

#cluster columns (ortholog groups) into groups with similar patterns
col_distance = dist(t(pangenome), method="binary")
col_cluster = hclust(col_distance, method="ward.D2")

#order the pangenome matrix from most common to most rare genes
pangenome_col <- pangenome[,order(colSums(pangenome),decreasing = TRUE)]


#make a table listing plasmid types for each strain in the same order as the pangenome matrix
plasmidtaxa<- data.frame(Isolate = plasmidtable$sample, PlasmidType = plasmidtable$subtype)
plasmidtaxa_ordered <- plasmidtaxa[match(row.names(pangenome), plasmidtaxa$Isolate),]
plasmidtypecolors <- as.data.frame(cbind(names(plasmidcolors),plasmidcolors)) 
colnames(plasmidtypecolors)<-c("type","color")
plasmidtaxa_ordered_color <- merge(plasmidtaxa_ordered,plasmidtypecolors,by.x="PlasmidType",by.y="type",all.x=TRUE)
plasmidcol_ordered <- plasmidtaxa_ordered_color[match(row.names(pangenome), plasmidtaxa_ordered_color$Isolate),]
plasmidcol<-plasmidcol_ordered$color
mycols<-cbind(as.character(plasmidcol),as.character(plasmidcol))


#plot the heatmap
heatmap.plus(as.matrix(pangenome_col), col = c("white", "blue"), Rowv = as.dendrogram(row_cluster), RowSideColors = mycols,scale="none", labCol=NA, Colv=NA, useRaster=TRUE)
legend(x="topleft",legend=plasmidtypecolors$type, fill=as.character(plasmidtypecolors$color), border=FALSE, bty="n", y.intersp = 0.7, cex=0.7)

```
